Spec.html

```diff
@@ -0,0 +1,252 @@
+        1. Let _nestedAssignmentPattern_ be the parse of the source text corresponding to |DestructuringAssignmentTarget| using either |AssignmentPattern| or |AssignmentPattern[Yield]| as the goal symbol depending upon whether this |AssignmentElement| has the <sub>[Yield]</sub> parameter.
+        1. Return the result of performing DestructuringAssignmentEvaluation of _nestedAssignmentPattern_ with _restObj_ as the argument.
+      </emu-alg>
+
+      <emu-grammar>ObjectAssignmentPattern : `{` AssignmentPropertyList `,` `...` DestructuringAssignmentTarget `}`</emu-grammar>
+      <emu-alg>
+        1. Let _excludedNames_ be the result of performing DestructuringAssignmentEvaluation for _AssignmentPropertyList_ using _value_ as the argument.
+        1. ReturnIfAbrupt(_excludedNames_).
+        1. If |DestructuringAssignmentTarget| is neither an |ObjectLiteral| nor an |ArrayLiteral|, then
+          1. Let _lref_ be the result of evaluating |DestructuringAssignmentTarget|.
+          1. ReturnIfAbrupt(_lref_).
+        1. Let _restObj_ be ObjectCreate(%ObjectPrototype%).
+        1. Let _assignStatus_ be CopyDataProperties(_restObj_, _value_, _excludedNames_).
+        1. ReturnIfAbrupt(_assignStatus_).
+        1. If |DestructuringAssignmentTarget| is neither an |ObjectLiteral| nor an |ArrayLiteral|, then
+          1. Return PutValue(_lref_, _restObj_).
+        1. Let _nestedAssignmentPattern_ be the parse of the source text corresponding to |DestructuringAssignmentTarget| using either |AssignmentPattern| or |AssignmentPattern[Yield]| as the goal symbol depending upon whether this |AssignmentElement| has the <sub>[Yield]</sub> parameter.
+        1. Return the result of performing DestructuringAssignmentEvaluation of _nestedAssignmentPattern_ with _restObj_ as the argument.
+      </emu-alg>
+    </ins>
+
+    <emu-note>The following are the same as the ES2015 DestructuringAssignmentEvaluation except it collects a list of all destructured property names rather than just empty completion.</emu-note>
+
+    <emu-grammar>AssignmentPropertyList : AssignmentPropertyList `,` AssignmentProperty</emu-grammar>
+    <emu-alg>
+      1. Let <del>_status_</del> <ins>_propertyNames_</ins> be the result of performing DestructuringAssignmentEvaluation for |AssignmentPropertyList| using _value_ as the argument.
+      1. ReturnIfAbrupt(<del>_status_</del> <ins>_propertyNames_</ins>).
+      1. <del>Return</del> <ins>Let _nextNames_ be</ins> the result of performing DestructuringAssignmentEvaluation for |AssignmentProperty| using _value_ as the argument.
+      1. <ins>ReturnIfAbrupt(_nextNames_).</ins>
+      1. <ins>Append each item in _nextNames_ to the end of _propertyNames_.</ins>
+      1. <ins>Return _propertyNames_.</ins>
+    </emu-alg>
+
+    <emu-grammar>AssignmentProperty : IdentifierReference Initializer?</emu-grammar>
+    <emu-alg>
+      1. Let _P_ be StringValue of |IdentifierReference|.
+      1. Let _lref_ be ? ResolveBinding(_P_).
```

Comment:
You had ReturnIfAbrupt but I converted to ? to match the current spec here and in CopyDataProperties.